src/utils/commands/command_util.py

```diff
@@ -202,17 +184,19 @@ def group_status(self) -> str:
             result += f"**Challenge {challenge['id']} | {date.strftime('%Y-%m-%d')}**\n\n"
             questions = self.database.weekly_questions.load_by_challenge_id(challenge["id"])
             user_scores = { user["leetcode_id"] : 0 for user in users }
-
             if len(questions) == 0:
                 result += "Current challenge is empty"
             else:
                 total_completions = 0
+                for user in users:
+                    self.update_user_completions(user["leetcode_id"], challenge["id"])
                 for question in questions:
                     completions = 0
                     for user in users:
-                        completed = self.leetcode.check_challenge_completion(
-                                    user["leetcode_id"],
-                                    question["title_slug"])
+                        completed = self.database.question_completions.check_completion(
+                            user["leetcode_id"],
+                            question["title_slug"]
+                            )
```

Comment:
This is good in that it eliminates redundant API calls. However, as it stands we make the database call `question_completions.check_completion()` for each question (q) and each user (u), which makes `O(q * u)` database calls. I think you can go ahead and call `question_completions.load_all_title_slugs_by_user()` in the `for user in users:` loop and populate a dictionary of completions for each user. Then in the `for question in questions:` loop we don't need any database calls.

Reply:
Great point, updated this section in the latest commit.